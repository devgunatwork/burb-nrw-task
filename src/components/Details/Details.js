import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import styles from './Details.module.scss';

/**
 * @description common component for details of show/episode
 * @param name: name of show/episode
 * @param summary: summary of show/episode
 * */
const Details = ({ name, summary }) => {
  return (
    <>
      <Typography className={styles.name} variant="h4" gutterBottom>
        {name}
      </Typography>
      <Typography variant="body1" gutterBottom>
        {summary}
      </Typography>
    </>
  );
};

Details.defaultProps = {
  name: '',
  summary: '',
};
Details.propTypes = {
  name: PropTypes.string,
  summary: PropTypes.string,
};

export default Details;
