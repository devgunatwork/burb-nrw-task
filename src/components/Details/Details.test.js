import React from 'react';
import Details from './Details';

describe('components', () => {
  describe('Details', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(<Details />);

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
