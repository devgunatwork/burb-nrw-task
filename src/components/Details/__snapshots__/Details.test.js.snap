// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Details UI should match the snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    className="name"
    gutterBottom={true}
    variant="h4"
  />
  <WithStyles(ForwardRef(Typography))
    gutterBottom={true}
    variant="body1"
  />
</Fragment>
`;
